migrate/doc.go

```diff
@@ -0,0 +1,37 @@
+/*
+Migrate handles translating the state of the blockchain between software versions.
+
+For example, as modules are changed over time the structure of the data they store changes. The data structure must be
+migrated to the new structure of the newer versions.
+
+There are two types of migration:
+- **genesis migration** that converts an exported genesis file from the old software to a new genesis file for the new software
+- **upgrade module migration** that converts data in the database itself from the old version to the new
```

Comment:
I'm not familiar with this type of upgrade. How does this even happen?


EDIT: I've realized you mean live upgrades here. I would recommend you use that nomenclature instead. Module migration isn't a thing really.

Reply:
cool, live upgrade is a better name